src/Microsoft.Extensions.Configuration.AzureAppConfiguration/Extensions/LoggingExtensions.cs

```diff
@@ -0,0 +1,75 @@
+﻿using Microsoft.Extensions.Logging;
+using System;
+using System.Text;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.Extensions
+{
+    internal static class LoggingExtensions
+    {
+        public static string FormatLog(string loggingConstant, string changeType = null, string key = null, string label = LabelFilter.Null, string endpoint = null)
+        {
+            StringBuilder fullLogMessage = new StringBuilder();
+            fullLogMessage.Append(loggingConstant);
+
+            if (changeType != null)
+            {
+                fullLogMessage.Append($" Change:'{changeType}'");
+            }
+
+            if (key != null)
+            {
+                fullLogMessage.Append($" Key:'{key}'");
+            }
+
+            if (label != LabelFilter.Null)
+            {
+                fullLogMessage.Append($" Label:'{label}'");
+            }
+
+            if (endpoint != null)
+            {
+                fullLogMessage.Append($" Endpoint:'{endpoint.TrimEnd('/')}'");
+            }
+
+            return fullLogMessage.ToString();
+        }
+
+        public static void HandleLog(ILogger logger, LogLevel logLevel, Exception e, string message)
```

Comment:
We should split it into LogWarning, LogInfo and LogDebug. Keeps the code easy to understand and avoids unnecessary input arguments (like Exception e), which are not applicable to 2 out of 3 log types here.